### ISSUE TITLE
Need to implement virtual scroll, preferably from the CDK

### ISSUE THREAD

**Issue by mflorence99** [mflorence99/el-file#2]
Even when directory already loaded, we may need virtual scroll!

**Comment by mflorence99** [comment]
We still need to have VS, but by far the big contributor to file load perf was in `layout/fs.ts`. The 'problem' with `ngx-electron` is that it makes it extremely easy to write code that you *think* is all running in one process. But of course any calls (say) to the file system are IPC'd to the main process. That's *very* expensive for directory loading and stat'ing. I moved all that very simply to the main process so that it is done all at once and Elf is now way faster.

Because objects are passed from one process to the other as JSON, the notion of a class with methods is lost. So an fs.Stats no longer has its methods when it is serialized. We lose some elegance in reifying but it is worth it.

**Comment by mflorence99** [comment]
I've implemented virtual scroll and it works well with very little code but for now I did not use the CDK. The difficulty is that Elf uses recursive descent -- a tree renders its root branch, then in turn it renders its rows and/or its child branches, and so on. In other words, there are multiple iterators within a scrolled viewport, whereas the CDK only contemplates one. 

Although @amcdnl strongly advised me to flatten out the recursion, the recursive code is so straightforward I was reluctant to do so at this stage.

So I used [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). It is easy to use and very fast. There's a visual anomaly I want to clean up before closing, though. I believe it would be so whatever technique I used. When sorting a huge set of data, the empty dehydrated rows are painted before they are rehydrated.
